src/orion/benchmark/task/task.py

```diff
@@ -0,0 +1,298 @@
+        if isinstance(item, Tensor):
+            item = item.cpu().numpy()
+        assert all(self.lower <= self.upper), "lower > upper?"
+        if not item.shape[-1] == self.lower.shape[-1]:
+            return False
+        return np.all(self.lower <= item) and np.all(item <= self.upper)
+
+
+class Task(BaseTask, Generic[HParams], ABC):
+    """ Simple ABC for a 'Task'.
+    
+    TODO: Ported this from the HPO-Warm-Start repo, could perhaps merge it into BaseTask
+
+    Can evaluate dataclasses and dicts, and can have some input dimensions be fixed by
+    passing them as keyword arguments to the constructor.
+    """
+
+    hparams: ClassVar[Type[HParams]]
+
+    def __init__(
+        self,
+        max_trials: int = 100,
+        task_id: int = 0,
+        rng: np.random.RandomState = None,
+        seed: int = None,
+        fixed_dims: Dict[str, Any] = None,
+    ):
+        super().__init__(max_trials=max_trials)
+        self.task_id = task_id
+        self.rng = rng or np.random.RandomState(seed)
+        self.seed = seed
+        fixed_dims = fixed_dims or {}
+        # Fixed dimensions in the hparam space.
+        self.fixed_values: Dict[str, Any] = {
+            key: value
+            for key, value in fixed_dims.items()
+            if key in self.hparams.get_orion_space_dict()
+        }
+        logger.debug(
+            f"Creating a Task of type {type(self).__name__}, with task id "
+            f"{task_id} and with fixed values {self.fixed_values}"
+        )
+
+    @abstractmethod
+    def __call__(
+        self, hp: Union[HParams, Dict, np.ndarray] = None, **kwargs
+    ) -> np.ndarray:
+        """Evaluates the given samples and returns the performance.
+
+        Args:
+            hp (Union[HyperParameters, Dict, np.ndarray], optional):
+                Either a Hyperparameter dataclass, a dictionary, or a numpy
+                array containing the values for each dimension. Defaults to
+                None.
+
+        Returns:
+            np.ndarray: The performances of the hyperparameter on the sampled
+            task.
+        """
+
+        # return super().__call__(**kwargs)
+
+    def call(self, *args, **kwargs):
+        return self(*args, **kwargs)
+
+    @property
+    def max_trials(self) -> int:
+        """Return the max number of trials to run with this objective."""
+        return self._max_trials
+
+    def get_search_space(self) -> Dict[str, str]:
+        """Return the search space for the task objective function"""
+        return self.space
+
+    @property
+    def configuration(self):
+        """Return the configuration of the task."""
+        return {
+            self.__class__.__qualname__: {
+                "task_id": self.task_id,
+                "seed": self.seed,
+                "fixed_values": self.fixed_values,
+                "max_trials": self.max_trials,
+            }
+        }
+
+    @property
+    def full_space(self) -> Dict[str, str]:
```

Comment:
Is there a difference between full_space and get_search_space?

Reply:
I figured it out here: https://github.com/Epistimio/orion/pull/632/files#r690685650